task/monorail-server/src/bsp/sidecar_bcd.rs

```diff
@@ -548,6 +585,31 @@ impl<'a, R: Vsc7448Rw> Bsp<'a, R> {
         let phy = vsc85xx::Phy::new(phy_port, &mut phy_rw);
         Some(callback(phy))
     }
+
+    pub(crate) fn unlock_vlans_until(
+        &mut self,
+        unlock_until: u64,
+    ) -> Result<(), MonorailError> {
+        ringbuf_entry!(Trace::UnlockUntil(unlock_until));
+        self.vsc7448
+            .configure_vlan_sidecar_unlocked()
+            .map_err(|e| {
+                ringbuf_entry!(Trace::UnlockError(e));
+                MonorailError::from(e)
+            })?;
+        self.vlan_mode = VLanMode::UnlockedUntil(unlock_until);
+        Ok(())
+    }
+
+    pub(crate) fn lock_vlans(&mut self) -> Result<(), MonorailError> {
+        ringbuf_entry!(Trace::LockingVLans);
+        self.vsc7448.configure_vlan_sidecar_locked().map_err(|e| {
+            ringbuf_entry!(Trace::LockError(e));
+            MonorailError::from(e)
+        })?;
+        self.vlan_mode = VLanMode::Locked;
```

Comment:
Same concern here.

Is there a way we could eliminate `vlan_mode` and ask the actual configured hardware whether we're locked or not, so they can't go out of sync?

Reply:
I'm on the fence here – yes, we can read back configuration registers, but they too could be in an inconsistent state – or the read could fail, since it's going to an external device!

I think the combination of

1) Fail towards the locked state
2) Return an error to the user if anything goes wrong (and the state could be inconsistent)

is okay for a first draft, since we'll be doing this manually and won't ignore the errors